lib/configDefaults.js

```diff
@@ -1,7 +1,30 @@
 const os = require('os')
 const path = require('path')
+const fs = require('fs')
 
-module.exports = {
+const reporter = require('./reporter')
+
+const userConfig = () => {
+    const configDir = path.join(os.homedir(), '.config', 'd2')
+    const configPattern = /^(d2\.)*(config)\.(js$|json$)/
```

Comment:
` /^(d2\.)*(config)\.(js)/` would be enough to capture both `json` and `js` files, but I opted for a more explicit variant that is easier to mentally parse.